release-notes/v3.31.0-release-notes.md

```diff
@@ -0,0 +1,204 @@
+- Enhanced IP Pool Allocation by introduce namespaceSelector Field [calico 10888](https://github.com/projectcalico/calico/pull/10888) (@PersistentJZH)
+- Scale: Kubernetes datastore: speed up IP allocation in clusters that have 1000+ IPAM blocks.  Previously, the IPAM plugin would list all block affinity resources for every request and filter client-side.  Add a label to the block affinity and filter on that instead, for efficient server-side filtering. [calico 10855](https://github.com/projectcalico/calico/pull/10855) (@fasaxc)
+- Add calicoctl validate command, which runs local validation against its input without applying changes to the datastore. [calico 10829](https://github.com/projectcalico/calico/pull/10829) (@Copilot)
+- Avoid loading wireguard kernel module when Wireguard is disabled. [calico 10821](https://github.com/projectcalico/calico/pull/10821) (@bartekzurawski)
+- Updates RBAC for kube-controllers to access services/servicestatus for ETCD manifest installation. [calico 10791](https://github.com/projectcalico/calico/pull/10791) (@MichalFupso)
+- Performance: Felix now loads BPF programs in parallel for reduced start-up latency. [calico 10771](https://github.com/projectcalico/calico/pull/10771) (@fasaxc)
+- Adds apiservices and webhooks information to the calicoctl diags bundle. [calico 10744](https://github.com/projectcalico/calico/pull/10744) (@rene-dekker)
+- You can now configure whether or not to aggregate routes for LoadBalancer services with externalTrafficPolicy=Cluster when advertised over BGP. [calico 10719](https://github.com/projectcalico/calico/pull/10719) (@PersistentJZH)
+- Set DSCP for workload's egress traffic leaving cluster [calico 10718](https://github.com/projectcalico/calico/pull/10718) (@mazdakn)
+- Cap whisker UI memory consumption [calico 10712](https://github.com/projectcalico/calico/pull/10712) (@ronanc-tigera)
+- Added typha configuration to allow server address to be configured [calico 10697](https://github.com/projectcalico/calico/pull/10697) (@Cellebyte)
+- Updates the HostEndpoint controller to be aware of all Node interfaces and IPs. Update the HostEndpoint template with interfaceSelector to be able to create automatic HostEndpoints containing interfaceName. Existing HostEndpoints created by the HostEndpoint controller might be updated after the
+- Calico LoadBalancerController will consider service.spec.LoadBalancerClass in determining if IP should be assigned. When in RequestedServicesOnly mode service will be assigned IP if LoadBalancerClass is set to calico. When running is AllServices and RequestedServicesOnly mode and the LoadBalancerClass is set to other value than calico, service will not be assigned an IP [calico 10627](https://github.com/projectcalico/calico/pull/10627) (@MichalFupso)upgrade to contain interfaces newly detected by Calico Node. [calico 10672](https://github.com/projectcalico/calico/pull/10672) (@MichalFupso)
```

Comment:
This bullet point for some reason got inserted in the middle of the entry directly above it. 

Fix it with the following: 
```suggestion
- Calico LoadBalancerController will consider service.spec.LoadBalancerClass in determining if IP should be assigned. When in RequestedServicesOnly mode service will be assigned IP if LoadBalancerClass is set to calico. When running is AllServices and RequestedServicesOnly mode and the LoadBalancerClass is set to other value than calico, service will not be assigned an IP [calico 10627](https://github.com/projectcalico/calico/pull/10627) (@MichalFupso)
```